### DISCUSSION TITLE
Checkbox

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#306] [Component suggestions]
<img width="480" alt="visuele weergave van de checkbox component" src="https://github.com/user-attachments/assets/17def6ab-94ec-43c0-9e52-ef0b656b8c22">

We willen graag meer leren over de Checkbox component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack]()

---

## Naam

Checkbox

## Doel
Invoerveld voor het kiezen van een optie uit een groep van opties, of om te kiezen tussen "wel" of "niet".

## Gerelateerde componenten
[Checkbox Group](https://github.com/orgs/nl-design-system/discussions/223), [Radio Button](https://github.com/orgs/nl-design-system/discussions/288), [Switch](https://github.com/orgs/nl-design-system/discussions/293)

## Onderzoek

[Checkboxes vs. Radio Buttons - Nielsen Norman Group](https://www.nngroup.com/articles/checkboxes-vs-radio-buttons/)
Artikel over verschil tussen Checkboxes en Radio Buttons.

[Inclusively Hiding & Styling Checkboxes and Radio Buttons - Sara Soueidan](https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/)
Artikel waarin Sara Soueidan beschrijft hoe je custom Checkboxes and Radio Buttons kunt ontwikkelen.

[We've updated the radios and checkboxes - GOV.UK](https://designnotes.blog.gov.uk/2016/11/30/weve-updated-the-radios-and-checkboxes-on-gov-uk/)
Tim Paul legt uit waarom GOV.UK aanpassingen heeft gedaan bij Radio en Checkbox opties.

[Checkboxes - GOV.UK](https://design-system.service.gov.uk/components/checkboxes/)
Onder 'Research on this component' worden bevindingen genoteerd. Er is ook een [link naar de Checkboxes Github discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/37) waar meer informatie, onderzoek en aanpassingen worden gedeeld.

In een [comment bij Checkbox Github Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/37#issuecomment-1576748073) wordt opgemerkt dat een high-contrast gebruiker gebaad is bij een visuele wijziging voor de hover state.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-checkbox--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29184)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/checkbox)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 09 51 44](https://github.com/user-attachments/assets/95c91991-54e4-4bbf-8d52-7de8920aeb72)

Hover
![Scherm­afbeelding 2024-09-30 om 09 52 33](https://github.com/user-attachments/assets/d738c329-ae35-4880-98de-b9f33a48d427)

Disabled
![Scherm­afbeelding 2024-09-30 om 10 00 13](https://github.com/user-attachments/assets/102eccd9-7d05-48d4-8590-f6c5a90c3a3e)

Focus
![Scherm­afbeelding 2024-09-30 om 09 53 51](https://github.com/user-attachments/assets/9a6ee09b-0383-4680-b2cb-6a7a625e2bd3)

Invalid
![Scherm­afbeelding 2024-09-30 om 10 00 32](https://github.com/user-attachments/assets/098f3038-1af7-4756-ab50-0d75815726be)

Checked
![Scherm­afbeelding 2024-09-30 om 09 52 54](https://github.com/user-attachments/assets/3d247103-5588-44e7-b140-0818f167d62c)

Checked + Hover
![Scherm­afbeelding 2024-09-30 om 09 52 59](https://github.com/user-attachments/assets/4b968404-c65d-4913-8117-108882213121)

Checked + Disabled
![Scherm­afbeelding 2024-09-30 om 10 00 20](https://github.com/user-attachments/assets/7e11a687-1cb2-4660-97f5-efffdce39671)

Checked + Invalid
![Scherm­afbeelding 2024-09-30 om 10 00 37](https://github.com/user-attachments/assets/5fbea0df-03a4-455d-bb4c-973dcafda446)

Indeterminate
![Scherm­afbeelding 2024-09-30 om 10 00 27](https://github.com/user-attachments/assets/760a8de2-0231-446f-adb6-bece5ff5ba38)

Long label
![Scherm­afbeelding 2024-09-30 om 09 51 49](https://github.com/user-attachments/assets/bacb23f5-d4e0-477a-b40a-36292cab1b68)

**Figma**

States
![Scherm­afbeelding 2024-09-30 om 09 54 44](https://github.com/user-attachments/assets/fbdd181e-ccbf-47db-9e10-ca44ab48bc25)



---

### Notities

- Documentatie beschikbaar (Storybook)
- Standaard met label.
- Label krijgt underline bij hover state.
- Optie voor Indeterminate.
- In Figma is er een mogelijkheid om een error-message onder de checkbox te tonen.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Amsterdam](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616509).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-input-checkbox--docs)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=1450-4570&t=acnBrMvzqeHWk9vR-4)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=944-1535)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Checkbox)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 12 24 27](https://github.com/user-attachments/assets/42ca59b2-48d2-4a09-9857-88b634568f60)

Checked
![Scherm­afbeelding 2024-09-30 om 12 24 32](https://github.com/user-attachments/assets/b42d38e1-dd11-4583-bb06-fd37c7e38df2)

Invalid
![Scherm­afbeelding 2024-09-30 om 12 24 36](https://github.com/user-attachments/assets/1006b533-4bc7-4a1f-94df-dcea65455798)

Disabled
![Scherm­afbeelding 2024-09-30 om 12 24 44](https://github.com/user-attachments/assets/8709251a-1236-4524-9acd-985aa0b69e18)

Hover
![Scherm­afbeelding 2024-09-30 om 12 24 48](https://github.com/user-attachments/assets/b23297ae-e211-4ec0-9c68-4be83b2dcaf3)

Focus
![Scherm­afbeelding 2024-09-30 om 12 24 52](https://github.com/user-attachments/assets/00a6e810-903d-422d-b34c-240b90ca8a9c)

Focus visible
![Scherm­afbeelding 2024-09-30 om 12 24 56](https://github.com/user-attachments/assets/b927d7db-ff46-4b98-b7d3-717389d0e165)

**Figma - Huidig**

Component varianten voor Checkbox
![Scherm­afbeelding 2024-09-30 om 12 26 52](https://github.com/user-attachments/assets/b1e8ba52-7fde-4d62-8139-64b1cb888679)

**Figma - WIP**

States van Checkbox als onderdeel van Form Field Checkbox Option
![Scherm­afbeelding 2024-02-28 om 12 12 16](https://github.com/nl-design-system/backlog/assets/545207/6e1a7a6f-1ec9-4b94-9b47-54f284aa513d)
![Scherm­afbeelding 2024-02-28 om 12 12 25](https://github.com/nl-design-system/backlog/assets/545207/44483a2d-3692-4b5e-b485-60e5fac8b382)
![Scherm­afbeelding 2024-02-28 om 12 12 32](https://github.com/nl-design-system/backlog/assets/545207/bb67b0b3-67d3-498a-9807-128cbc0da946)

---

### Notities

- Storybook en Figma (WIP) zonder label. Figma (Huidig) standaard met label.
- [Zie comment van Rozerin](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8593297) over onderzoek waarbij grotere checkboxes werden ingezet.
- Figma - Huidig component heeft ook een mogelijkheid voor 'Large' variant. Hierbij schaalt het label mee.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Den Haag](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616643).

**Comment by jeffreylauwers** [comment]
## DSO

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-selectable--checkbox)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 12 44 09](https://github.com/user-attachments/assets/0caaf0e6-67b8-4527-a685-b3e5029dac5f)

Active
![Scherm­afbeelding 2024-09-30 om 12 44 50](https://github.com/user-attachments/assets/23988e51-0b68-48a1-999f-21fa2b234cce)

Focus
![Scherm­afbeelding 2024-09-30 om 12 44 58](https://github.com/user-attachments/assets/c7fa0793-9f85-470d-84e8-adb56103c4f6)

Checked
![Scherm­afbeelding 2024-09-30 om 12 44 24](https://github.com/user-attachments/assets/8aba7e23-c604-4536-88da-cdf316c802e7)

---

### Notities

- Standaard met label.
- Optie om een 'i-tje' toe te voegen achter het label.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van DSO](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616729).

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/checkbox)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-09-30 om 13 03 45](https://github.com/user-attachments/assets/a7828b0e-ff91-4104-9d4f-626d049337bf)

Focus
![Scherm­afbeelding 2024-09-30 om 13 04 28](https://github.com/user-attachments/assets/59d0c5f7-9854-480b-b845-946b89d24400)

Disabled
![Scherm­afbeelding 2024-09-30 om 13 04 47](https://github.com/user-attachments/assets/30324b30-094c-4985-99f4-e17e776c3ac2)

Checked
![Scherm­afbeelding 2024-09-30 om 13 03 57](https://github.com/user-attachments/assets/db2df877-f459-4992-a6dd-daf3fd4ec042)

Ook gebruikt vooraf aan een table-row zonder label. Hierbij wijken de states af.

Default
![Scherm­afbeelding 2024-09-30 om 13 05 21](https://github.com/user-attachments/assets/a3b402be-8c56-4404-b68d-e8bc77fc08b5)

Checked en Indeterminate state
![Scherm­afbeelding 2024-09-30 om 13 05 31](https://github.com/user-attachments/assets/8931ab7f-c2ca-49b0-baf1-bda2c6cb784e)

Focus
![Scherm­afbeelding 2024-09-30 om 13 05 44](https://github.com/user-attachments/assets/ecba7193-ae09-468f-b25f-74f7c44208b2)

---

### Notities
- Documentatie beschikbaar.
- Standaard met label. Maar kan ook zonder label zoals te zien in het voorbeeld met Table.
- Kader om checkbox en label, in het geheel 'clickable'.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van DUO](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616846).

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-checkbox--default)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 14 23 48](https://github.com/user-attachments/assets/6038af97-db9a-4d8d-b5b5-f9afb8aa0992)

Focus
![Scherm­afbeelding 2024-09-30 om 14 24 35](https://github.com/user-attachments/assets/490b4dc5-84a1-4627-9785-91fb0c0a4ad6)

Checked
![Scherm­afbeelding 2024-09-30 om 14 25 00](https://github.com/user-attachments/assets/53b1b19b-cabd-481f-a458-9c0196de5024)

Required
![Scherm­afbeelding 2024-10-01 om 09 05 04](https://github.com/user-attachments/assets/87a8c9a5-75a2-469a-82a8-98d02775d104)


---

### Notities

- Standaard met label.
- Kader om checkbox en label, in het geheel 'clickable'.
- Mogelijkheid om Checkbox 'Required' te maken waarbij er een rode astrix wordt toegevoegd.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Open Formulieren](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8592449).

**Comment by jeffreylauwers** [comment]
## OpenGemeenten
[Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

[Checkbox](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-Checkbox-Checkbox/01-molecules-080-Forms-And-Search-Checkbox-Checkbox.html)

Default
![Scherm­afbeelding 2024-09-30 om 14 30 42](https://github.com/user-attachments/assets/cb50f670-70dc-43c9-ada4-0ff19d2e6870)

Hover
![Scherm­afbeelding 2024-09-30 om 14 31 19](https://github.com/user-attachments/assets/60c360d9-0fdb-4128-b8b5-c2cb20a937b6)

Focus
![Scherm­afbeelding 2024-09-30 om 14 33 48](https://github.com/user-attachments/assets/e88508df-f73d-45b3-acdf-fa6d4885a868)

Checked
![Scherm­afbeelding 2024-09-30 om 14 31 39](https://github.com/user-attachments/assets/57aaadab-1141-4658-b5a1-3f938c570dfa)

Checked + Hover
![Scherm­afbeelding 2024-09-30 om 14 31 56](https://github.com/user-attachments/assets/75ad8a4f-d814-461f-8c76-c112269fd814)

[Checkbox Description](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Description/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Description.html)
![Scherm­afbeelding 2024-09-30 om 14 30 52](https://github.com/user-attachments/assets/d33246c7-d1d3-4e40-955b-0e16a5fc4de1)

[Checkbox Error](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Error/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Error.html)
![Scherm­afbeelding 2024-09-30 om 14 30 59](https://github.com/user-attachments/assets/1e7214e2-9324-41de-82d2-262ca5c3b813)

[Checkbox Required](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Required/01-molecules-080-Forms-And-Search-Checkbox-Checkbox-Required.html)
![Scherm­afbeelding 2024-09-30 om 14 31 09](https://github.com/user-attachments/assets/1f6e5f4a-bc52-4310-bd57-e6cfe3c599b2)


---

### Notities
- Standaard met label.
- Form field checkbox option is ook voorbij het zichtbare label nog klikbaar.
- Mogelijkheid beschikbaar voor extra beschrijving.
- Mogelijkheid beschikbaar voor validatie en foutmelding.
- Mogelijkheid beschikbaar om aan te duiden dat de optie verplicht is.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van OpenGemeenten](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616917).

**Comment by jeffreylauwers** [comment]
## Rotterdam

[Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=521-28182&mode=design&t=B6BdewKoEAslWiOU-4)

---

### Screenshots / Varianten

**Figma**

States
![Scherm­afbeelding 2024-02-28 om 11 57 46](https://github.com/nl-design-system/backlog/assets/545207/18724c93-8ac4-47b1-8d89-f4bf56d2ad52)


---

### Notities
- Onder de noemer: Toggles
- Standaard met label.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Rotterdam](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616968).

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-checkbox--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=109-11444&t=UUySCSQV8cCJHOd0-4)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-checkbox)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 14 47 47](https://github.com/user-attachments/assets/db469055-fdd1-4f1a-910c-41ac169c2204)

Hover
![Scherm­afbeelding 2024-09-30 om 14 47 55](https://github.com/user-attachments/assets/65338d3a-2916-47a5-83f6-57d8bd1cd222)

Active
![Scherm­afbeelding 2024-09-30 om 14 48 17](https://github.com/user-attachments/assets/86f16d4a-c39f-4f12-9764-048d888eb5fd)

Disabled
![Scherm­afbeelding 2024-09-30 om 14 48 06](https://github.com/user-attachments/assets/2a43ee7f-29c5-43cd-8c29-7685771b7e31)

Focus
![Scherm­afbeelding 2024-09-30 om 14 48 24](https://github.com/user-attachments/assets/df85bdf9-3a92-4845-ac38-102ca3fcc888)

Indeterminate
![Scherm­afbeelding 2024-09-30 om 14 48 35](https://github.com/user-attachments/assets/0a16e96a-7356-46b1-a40e-1e2c666b3c87)

Invalid
![Scherm­afbeelding 2024-09-30 om 14 48 48](https://github.com/user-attachments/assets/1ba56eea-cf1b-4d45-923c-5c21e8acd081)

Checked
![Scherm­afbeelding 2024-09-30 om 14 49 06](https://github.com/user-attachments/assets/7ff7bb90-6594-4faa-9894-c166d7403f1e)

Checked + Hover
![Scherm­afbeelding 2024-09-30 om 14 49 13](https://github.com/user-attachments/assets/e2b02c49-26fe-4960-bf41-c2e112355490)

**Figma**
![Scherm­afbeelding 2024-09-30 om 14 49 29](https://github.com/user-attachments/assets/99a3516e-d253-42b4-a2e7-f10d41284dbf)



---

### Notities

- Standaard met label.
- Mogelijkheid voor Indeterminate state.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van RVO](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8617018).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-checkbox--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=944-1535)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/checkbox)

---

### Screenshots / Varianten

**Storybook**

Veel verschillende states beschikbaar in Storybook. Maar niet elke state zorgt voor een wisseling in vormgeving. Vandaar dat er maar enkele states meegenomen in screenshots.

Default
![Scherm­afbeelding 2024-09-30 om 15 26 35](https://github.com/user-attachments/assets/43f836fa-6c9d-4512-a112-3e330185ede8)

Focus
![Scherm­afbeelding 2024-09-30 om 15 26 50](https://github.com/user-attachments/assets/32663064-d2be-472c-bd18-8a08aebf3d15)

Disabled
![Scherm­afbeelding 2024-09-30 om 15 26 56](https://github.com/user-attachments/assets/a6df2bec-7c3b-420a-8660-ef52531a9bd0)

Checked
![Scherm­afbeelding 2024-09-30 om 15 27 04](https://github.com/user-attachments/assets/467d0719-53d9-4b57-aa4d-98bec70b9523)

Checked + Disabled
![Scherm­afbeelding 2024-09-30 om 15 27 12](https://github.com/user-attachments/assets/9391cdc5-7306-4748-8108-095bc9ffe43d)

Indeterminate
![Scherm­afbeelding 2024-09-30 om 15 27 23](https://github.com/user-attachments/assets/4f6ce681-b274-4adf-88eb-9422f3546e5b)

Invalid
![Scherm­afbeelding 2024-09-30 om 15 27 29](https://github.com/user-attachments/assets/fd6fea44-2b87-4e3b-9300-613c485c4860)

States van Checkbox als onderdeel van Form Field Checkbox Option
![Scherm­afbeelding 2024-02-28 om 12 11 39](https://github.com/nl-design-system/backlog/assets/545207/7045cf1e-690c-4bdd-93ed-84b5a392cbbf)
![Scherm­afbeelding 2024-02-28 om 12 11 49](https://github.com/nl-design-system/backlog/assets/545207/55ecbbfc-47b6-49e1-8713-6b0bf6b29e61)
![Scherm­afbeelding 2024-02-28 om 12 11 55](https://github.com/nl-design-system/backlog/assets/545207/f96f8bc2-1ddf-4c08-804b-6267d1a69494)


---

### Notities

- Documentatie beschikbaar (Storybook)
- Zonder label. 
- Mogelijkheid voor Indeterminate state.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Utrecht](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8617136).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Checkbox op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=checkbox+%2364)

**Comment by Yolijn** [comment]
## ✨ Dit component is nu Community ✨


Help je mee hem door de Candidate stappen te krijgen?
[Checkbox op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=%2364)
